hugegraph-api/src/main/java/com/baidu/hugegraph/license/LicenseVerifyManager.java

```diff
@@ -177,20 +175,40 @@ private void checkIpAndMac(ExtraParam param) {
         if (StringUtils.isEmpty(expectMac)) {
             return;
         }
-        String actualMac;
-        try {
-            actualMac = this.machineInfo.getMacByInetAddress(
-                        InetAddress.getByName(expectIp));
-        } catch (UnknownHostException e) {
-            throw newLicenseException(
-                  "Failed to get mac address for ip '%s'", expectIp);
-        }
-        String expectFormatMac = expectMac.replaceAll(":", "-");
-        String actualFormatMac = actualMac.replaceAll(":", "-");
-        if (!actualFormatMac.equalsIgnoreCase(expectFormatMac)) {
-            throw newLicenseException(
-                  "The server's mac '%s' doesn't match the authorized '%s'",
-                  actualMac, expectMac);
+
+        // Mac must not empty
+        if (!StringUtils.isEmpty(expectIp)) {
+            String actualMac;
+            try {
+                actualMac = this.machineInfo.getMacByInetAddress(
+                            InetAddress.getByName(expectIp));
+            } catch (UnknownHostException e) {
+                throw newLicenseException(
+                      "Failed to get mac address for ip '%s'", expectIp);
+            }
+            String expectFormatMac = expectMac.replaceAll(":", "-");
+            String actualFormatMac = actualMac.replaceAll(":", "-");
+            if (!actualFormatMac.equalsIgnoreCase(expectFormatMac)) {
+                throw newLicenseException(
+                      "The server's mac '%s' doesn't match the authorized '%s'",
+                      actualMac, expectMac);
+            }
+        } else {
+            matched = false;
+            String expectFormatMac = expectMac.replaceAll(":", "-");
+            List<String> actualMacs = this.machineInfo.getMacAddress();
+            for (String actualMac : actualMacs) {
+                String actualFormatMac = actualMac.replaceAll(":", "-");
```

Comment:
why not keep ":"

Reply:
There may be two kinds of join symbols in the mac address, ':' and '-', they are both valid. To avoid check failures due to symbol mismatches, so unified the symbol to '-'